pkg/model/resource/resource.go

```diff
@@ -67,3 +70,20 @@ func (r *Resource) GVK() config.GVK {
 		Kind:    r.Kind,
 	}
 }
+
+func wrapKey(key string) string {
+	return fmt.Sprintf("%%[%s]s", key)
+}
+
+func (r Resource) Replacer() *strings.Replacer {
+	replacements := make([]string, 0, 10)
```

Comment:
Using a bounded slice is good for optimization when performance is an issue. Since performance isn't an issue this case, preferring maintainable code over performant code in this case, we should switch to unbounded slice (nil) in this case.